src/main/java/com/aws/greengrass/authorization/AuthorizationHandler.java

```diff
@@ -56,6 +60,7 @@
 public class AuthorizationHandler  {
     public static final String ANY_REGEX = "*";
     public static final String SECRETS_MANAGER_SERVICE_NAME = "aws.greengrass.SecretManager";
+    public static final String SHADOW_MANAGER_SERVICE_NAME = "aws.greengrass.ShadowManager";
```

Comment:
I'd strongly suggest that you consider rewriting this logic so that services can be registered dynamically.

Reply:
This was an issue raised by @rbattle before. I don't know how long this effort would take but this can be done in a future PR. In the interest of time we can table that and push this through to unblock ShadowManager.

Reply:
yes - we've talked with Joan already about the need to have dynamic registration support in the nucleus. Secret Manager also shouldn't be hardcoded in here. This needs to be addressed - but it requires some more design on the plugin interface.